src/app/add-clients/pageComponents.tsx

```diff
@@ -0,0 +1,740 @@
+                });
+            } else if (nonZeroAdultEntry.length === 0) {
+                setErrorMessages({
+                    ...errorMessages,
+                    numberAdultsErrorMessage: "This is a required field.",
+                });
+            } else {
+                setErrorMessages({
+                    ...errorMessages,
+                    numberAdultsErrorMessage: "",
+                });
+            }
+        };
+    };
+
+    const getNumberChildren: OnChangeType = (event) => {
+        const input = event.target.value;
+        const numberPattern = /^\d+$/;
+
+        if (input === "") {
+            setErrorMessages({
+                ...errorMessages,
+                numberChildrenErrorMessage: "This is a required field.",
+            });
+        } else if (input.match(numberPattern)) {
+            setErrorMessages({ ...errorMessages, numberChildrenErrorMessage: "" });
+            setNumberChildren(parseInt(input));
+        } else {
+            setErrorMessages({
+                ...errorMessages,
+                numberChildrenErrorMessage: "Please enter a valid number.",
+            });
+        }
+    };
+
+    const getGenderChildren = (count: number): OnChangeType => {
+        return (event) => {
+            const input = event.target.value !== "don't know" ? event.target.value : "child";
+            const particularChild = ageGenderChildren.findIndex((object) => object.key === count);
+            ageGenderChildren[particularChild].gender = input as PersonType;
+            setAgeGenderChildren([...ageGenderChildren]);
+        };
+    };
+
+    const getAgeChildren = (count: number): OnChangeType => {
+        return (event) => {
+            const input = event.target.value;
+            const particularChild = ageGenderChildren.findIndex((object) => object.key === count);
+            ageGenderChildren[particularChild].age =
+                input !== "Don't Know" ? parseInt(input) : null;
+            setAgeGenderChildren([...ageGenderChildren]);
+        };
+    };
+
+    const getBabyProducts: OnChangeType = (event) => {
+        const input = event.target.value;
+        if (input === "Yes") {
+            setErrorMessages({ ...errorMessages, nappyErrorMessage: "N/A" });
+            setBabyProducts(true);
+        } else {
+            setErrorMessages({ ...errorMessages, nappyErrorMessage: "" });
+            if (input === "No") {
+                setBabyProducts(false);
+            } else {
+                setBabyProducts(null);
+            }
+        }
+    };
+
+    const getNappySize: OnChangeType = (event) => {
+        const input = event.target.value;
+        if (input === "") {
+            setErrorMessages({ ...errorMessages, nappyErrorMessage: "This is a required field." });
+        } else {
+            setErrorMessages({ ...errorMessages, nappyErrorMessage: "" });
+            setNappySize(input);
+        }
+    };
```

Comment:
see if some of these `get*` functions are able to generalised and unified into fewer functions.

Reply:
Thanks. We missed this. `getNappySize` can be replaced by `getRequiredFieldWithoutChecks`. 

However, for the other functions, the only one that we think can be combined (easily) is `getAddressPostcode` and `getPhoneNumber`. However, doing that doesn't really make the code any more readable or shorter. 

We can show you how we would do it (i.e. the code) and you can give some feedback on that if necessary?